### DISCUSSION TITLE
Mail Server Buildpack

### DISCUSSION THREAD

**Discussion by schmijos** [paketo-buildpacks/feedback#214] [Ideas]
Wouldn't it be nice to be in control of my own little mail server if I need to send only few emails?

If there was a buildpack I'd use it to send transactional emails from my web apps.
I would assume that this should only work for not-shared IP addresses and would exclude marketing emails, because this would be way too hard. But actually why should the buildpack care? It only has to make sane defaults and help you to set up DMARC, SPF and DKIM, etc.

Does this make sense? Is this achievable?

**Comment by robdimsdale** [comment]
It sounds like you're asking for a buildpack to install and configure a mail server application? If so, that's definitely achievable. I'd suggest you start by reading the [buildpack author guide](https://buildpacks.io/docs/buildpack-author-guide/create-buildpack/).

This is unlikely to be something we would support in paketo, but that doesn't stop you from making your own!

**Reply by robdimsdale** [reply]
I should also clarify, you could buildpacks to create an image for the buildpack server, as well as provide client libraries.

But I'm not sure I'd advocate using buildpacks for the former - it seems simpler to use Dockerfiles directly because the primary advantages of buildpacks are secure, sensible defaults for turning code into applications and ease of composability, and neither would apply to installing a standalone application.

For the second use-case - providing client libraries and providing configuration for applications - that's definitely a good use for buildpacks.
